Python/Product/ProjectWizards/MsPythonConfigWizard.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Windows.Forms;
+using EnvDTE;
+using EnvDTE80;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.TemplateWizard;
+
+namespace Microsoft.PythonTools.ProjectWizards {
+    public sealed class MsPythonConfigWizard : IWizard {
+        private const string _configFileName = "mspythonconfig.json";
+
+        public void BeforeOpeningFile(ProjectItem projectItem) { }
+        public void ProjectFinishedGenerating(Project project) { }
+        public void ProjectItemFinishedGenerating(ProjectItem projectItem) { }
+
+        public void RunFinished() { }
+
+        public void RunStarted(object automationObject, Dictionary<string, string> replacementsDictionary, WizardRunKind runKind, object[] customParams) {
+            if (!(automationObject is DTE2 dte)) {
+                return;
+            }
+
+            var projectPath = TryGetProjectPath(dte);
+            if (string.IsNullOrEmpty(projectPath)) {
+                replacementsDictionary.TryGetValue("$solutiondirectory$", out projectPath);
+            }
+
+            if (string.IsNullOrEmpty(projectPath)) {
+                return;
+            }
+
+            var fileName = Path.Combine(projectPath, _configFileName);
+            if (File.Exists(fileName)) {
+                MessageBox.Show(Strings.ConfigFileAlreadyExists, Strings.ProductTitle, MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1);
+            } else {
+                File.WriteAllText(fileName,
```

Comment:
This might throw if for example if folder is not writeable by user, is it handled by the IDE or do we have to take care of it?

Reply:
I guess IDE since I copied it from EditorConfig wizard. Wizards are third party extensions so IDE should be try-catching all calls.